pkg/controllers/dynakube/version/activeGateVersionReconciler.go

```diff
@@ -0,0 +1,96 @@
+	case dynakube.FeaturePublicRegistry():
+		result.Source = status.PublicRegistryVersionSource
+	default:
+		result.Source = status.TenantRegistryVersionSource
+	}
+	return result
+}
+
+// Reconcile initializes state used for reconciliation and updates the version status used by the dynakube
+func (reconciler *ActiveGateImageVersionReconciler) Reconcile(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) error {
+	current := initializeVersionStatus(dynakube, reconciler.timeProvider.Now())
+	if reconciler.needsUpdate(current, dynakube) {
+		updatedStatus, err := reconciler.run(ctx, current, dynakube)
+		if err != nil {
+			return err
+		}
+		dynakube.Status.ActiveGate.VersionStatus = updatedStatus
+	}
+	return nil
+}
+
+func (reconciler *ActiveGateImageVersionReconciler) needsUpdate(current status.VersionStatus, dynakube *dynatracev1beta1.DynaKube) bool {
+	previous := dynakube.Status.ActiveGate.VersionStatus
+	if !dynakube.NeedsActiveGate() {
+		log.Info("skipping version status update for disabled section", "updater", "activegate")
+		return false
+	}
+
+	if current.Source != previous.Source {
+		log.Info("source changed, update for version status is needed", "updater", "activegate")
+		return true
+	}
+
+	if current.CustomImageNeedsReconciliation(logCustomActiveGateImageChangedMessage(current, previous), dynakube.CustomActiveGateImage()) {
+		return true
+	}
+
+	return isOutdated(logSkipUpdateTimestampValidMessage("activegate"), reconciler.timeProvider, dynakube, dynakube.Status.ActiveGate.LastProbeTimestamp)
+}
+
+func logCustomActiveGateImageChangedMessage(current status.VersionStatus, previous status.VersionStatus) func() {
+	return func() {
+		log.Info("custom image value changed, update for version status is needed", "updater", "activegate", "oldImage", previous.ImageID, "newImage", current.ImageID)
+	}
+}
+
+func (reconciler *ActiveGateImageVersionReconciler) run(ctx context.Context, current status.VersionStatus, dynakube *dynatracev1beta1.DynaKube) (status.VersionStatus, error) {
+	previous := dynakube.Status.ActiveGate.VersionStatus
+	if current.Source == status.CustomImageVersionSource {
+		log.Info("updating version status according to custom image", "updater", "activegate")
+
+		return updateVersionWithImage(ctx, reconciler.registryClient, current, dynakube.CustomActiveGateImage())
+	}
+
+	if !dynakube.FeatureDisableActiveGateUpdates() {
+		if InitialUpdateInProgress(previous) {
+			log.Info("initial status update in progress with no auto update", "updater", "activegate")
+		} else if current.Source == previous.Source {
+			log.Info("status updated skipped, due to no auto update", "updater", "activegate")
+			return previous, nil
+		}
+	}
+
+	imageUri, err := getImageUri(dynakube, reconciler.dtClient.GetLatestActiveGateImage, dynakube.DefaultActiveGateImage())
+	if err != nil {
+		return previous, err
+	}
+	return updateVersionWithImage(ctx, reconciler.registryClient, current, imageUri)
```

Comment:
you are not checking here about the public-registry, which would be nice if it worked but

`updateVersionWithImage` uses the digest in the ImageID which should not be used with the tenant registry (the digest changes there overtime)